Tests/AttributedStringStyleTests.swift

```diff
@@ -555,6 +555,134 @@ class StringStyleTests: XCTestCase {
         XCTAssertFalse(stillHasAltSixDict)
     }
 
+    func testStyleBlockRules() {
+        let string = "0<one attr1=\"11\">1<two attr2=\"12\">2</two></one>"
+
+        var tagAttr1Value: String!
+        var tagAttr2Value: String!
+
+        let rules: [XMLStyleRule] = [
+            .styleBlock("one") { attributes in
+                guard let val1 = attributes["attr1"], let intVal1 = Int(val1) else {
+                    XCTFail("attr1 not found")
+                    return StringStyle()
+                }
+
+                tagAttr1Value = val1
```

Comment:
I think you may be able to simplify this test. Rather than asserting things inside the closure, do some work in the closure, and afterward, asset that the result is as expected. Does that work for you, or would it miss some aspect of the way the closers work that you want to test?

Reply:
I pushed another commit where all `guard`s are removed. A bit better, I think. Still, lots of typecasts and `??`...